p2p/background/router.go

```diff
@@ -166,6 +195,219 @@ func (rtr *backgroundRouter) RemovePeer(peer typesP2P.Peer) error {
 	return rtr.pstore.RemovePeer(peer.GetAddress())
 }
 
+// setupUnicastRouter configures and assigns `rtr.UnicastRouter`.
+func (rtr *backgroundRouter) setupUnicastRouter() error {
+	unicastRouterCfg := config.UnicastRouterConfig{
+		Logger:         rtr.logger,
+		Host:           rtr.host,
+		ProtocolID:     protocol.BackgroundProtocolID,
+		MessageHandler: rtr.handleBackgroundMsg,
+		PeerHandler:    rtr.AddPeer,
+	}
+
+	unicastRouter, err := unicast.Create(rtr.GetBus(), &unicastRouterCfg)
+	if err != nil {
+		return fmt.Errorf("setting up unicast router: %w", err)
+	}
+
+	rtr.UnicastRouter = *unicastRouter
+	return nil
+}
+
+func (rtr *backgroundRouter) setupDependencies(ctx context.Context, cfg *config.BackgroundConfig) error {
+	if err := rtr.setupUnicastRouter(); err != nil {
+		return err
+	}
+
+	if err := rtr.setupPeerDiscovery(ctx); err != nil {
+		return fmt.Errorf("setting up peer discovery: %w", err)
+	}
+
+	if err := rtr.setupPubsub(ctx); err != nil {
+		return fmt.Errorf("setting up pubsub: %w", err)
+	}
+
+	if err := rtr.setupTopic(); err != nil {
+		return fmt.Errorf("setting up topic: %w", err)
+	}
+
+	if err := rtr.setupSubscription(); err != nil {
+		return fmt.Errorf("setting up subscription: %w", err)
+	}
+
+	if err := rtr.setupPeerstore(
+		ctx,
+		cfg.PeerstoreProvider, cfg.CurrentHeightProvider); err != nil {
```

Comment:
If we're multi-lining args, we gotta do it right

```suggestion
		cfg.PeerstoreProvider,
		 cfg.CurrentHeightProvider,
	 ); err != nil {
```

Reply:
Thanks for catching this! This change got caught in-between two branches and didn't quite make it all the way into either. :sweat_smile: